frontend/src/tree/TreeItem.tsx

```diff
@@ -78,6 +114,38 @@ export const TreeItem = ({
   setSelection,
   readOnly,
 }: TreeItemProps) => {
+  const classes = useTreeItemStyle();
+
+  const ItemCollapseToggle = ({ item, depth, onExpand, dataTestid }) => {
+    if (item.hasChildren) {
+      const onClick = () => onExpand(item.id, depth);
+      if (item.expanded) {
+        return (
+          <ArrowExpanded
+            title="Collapse"
+            className={classes.arrow}
+            width="20"
+            height="20"
+            onClick={onClick}
+            data-testid={dataTestid}
+          />
+        );
+      } else {
+        return (
+          <ArrowCollapsed
+            title="Expand"
+            className={classes.arrow}
+            width="20"
+            height="20"
+            onClick={onClick}
+            data-testid={dataTestid}
+          />
+        );
+      }
+    }
+    return <div></div>;
+  };
+
```

Comment:
No need to move this code, the composent gets re-created every time a refresh is performed. That has a cost.